File: node/zkgroup/auth/AuthCredential.ts
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import ByteArray from '../internal/ByteArray';
import NativeImpl from '../../NativeImpl';

export default class AuthCredential extends ByteArray {
  static SIZE = 181;

  constructor(contents: Buffer) {
    super(contents, AuthCredential.SIZE, true);
    NativeImpl.AuthCredential_CheckValidContents(contents);
  }
}

File: java/shared/java/org/signal/zkgroup/auth/AuthCredential.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.zkgroup.auth;

import org.signal.zkgroup.InvalidInputException;
import org.signal.zkgroup.internal.ByteArray;
import org.signal.client.internal.Native;

public final class AuthCredential extends ByteArray {
  public AuthCredential(byte[] contents) throws InvalidInputException {
    super(contents);
    try {
      Native.AuthCredential_CheckValidContents(contents);
    } catch (IllegalArgumentException e) {
      throw new InvalidInputException(e.getMessage());
    }
  }
}
